junit-jupiter-params/src/main/java/org/junit/jupiter/params/converter/ArgumentConverter.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.jupiter.params.converter;
+
+import org.junit.jupiter.api.extension.ParameterContext;
+
+public interface ArgumentConverter {
```

Comment:
What would we `S` and `T` for `JavaTimeArgumentConverter` and `DefaultArgumentConverter`? Both `Object`?

If we add generics we will have to use unchecked calls to the converter and make sure we catch resulting `ClassCastExceptions` in `ParameterizedTestParameterResolver`. I'm not sure that makes it more readable.